secondhw/src/main/io/codelex/java_advanced_test/exercise2/BasketApp.java

```diff
@@ -0,0 +1,26 @@
+package io.codelex.java_advanced_test.exercise2;
+
+public class BasketApp {
+    public static void main(String[] args) {
```

Comment:
Dēļ kļūdām, kuras pieļautas pašā Basket šeit sanāk diezgan liela putra:

```
Apple apple = new Apple();
Mushroom mushroom = new Mushroom();
```
<= šis ir ok, tiek izveidots ābols un sēne
```
Basket<Apple> appleBasket = new Basket<>(apple);
Basket<Mushroom> mushroomBasket = new Basket<>(mushroom);
```
<= šis ir gandrīz ok, jo tiek izvedoti jauni grozi, bet nav ok, ka viņiem uzreiz tiek padots ābols un sēne.
Un tad sākas lietas, kas pilnībā nav pareizas:
`Basket.addToBasket(); ` <= šis vēršas pie klases metodes addToBasket, nevis pie kāda konkrēta no klases izveidota objekta, piemēram appleBasket vai mushroomBasket, bet gan pie pašas klases un tā metode rediģē klases stāvokli (static int numberOfItems; un tas nav objektu, jebšu izveidoto grozu stāvoklis) un tas pilnībā nav pareizi. Nevienā vietā, Tu nepapildini appleBasket ar vēl kādu apple objektu vai mushroomBasket ar vēl kādu Mushroom objektu. Lai tos papildinātu, būtu jāsauc konkrētam objektam tā metode:
`appleBasket.addToBasket(apple)`, bet lai tas strādātu Basket klase ir jāsalabo.